libs/dev-dock/backend/src/backend.test.ts

```diff
@@ -0,0 +1,167 @@
+  featureFlagMock.disableFeatureFlag(
+    BooleanEnvironmentVariableName.ENABLE_DEV_DOCK
+  );
+  const { apiClient } = setup();
+  await expect(apiClient.getElection()).rejects.toThrow();
+  await expect(apiClient.getUsbDriveStatus()).rejects.toThrow();
+  await expect(apiClient.getCardStatus()).rejects.toThrow();
+});
+
+// Note: This test overwrites the global mock card state.
+test('card mock endpoints', async () => {
+  const { apiClient } = setup();
+  await apiClient.removeCard(); // Reset card state to no_card to start in case it's not already
+
+  await expect(apiClient.getCardStatus()).resolves.toEqual({
+    status: 'no_card',
+  });
+
+  await apiClient.insertCard({ role: 'system_administrator' });
+  await expect(apiClient.getCardStatus()).resolves.toEqual({
+    status: 'ready',
+    cardDetails: {
+      user: fakeSystemAdministratorUser({ jurisdiction: DEV_JURISDICTION }),
+    },
+  });
+
+  await apiClient.removeCard();
+  await expect(apiClient.getCardStatus()).resolves.toEqual({
+    status: 'no_card',
+  });
+
+  await apiClient.insertCard({ role: 'election_manager' });
+  await expect(apiClient.getCardStatus()).resolves.toEqual({
+    status: 'ready',
+    cardDetails: {
+      user: fakeElectionManagerUser({
+        electionHash: electionSampleDefinition.electionHash,
+        jurisdiction: DEV_JURISDICTION,
+      }),
+    },
+  });
+
+  await apiClient.removeCard();
+
+  await apiClient.insertCard({ role: 'poll_worker' });
+  await expect(apiClient.getCardStatus()).resolves.toEqual({
+    status: 'ready',
+    cardDetails: {
+      user: fakePollWorkerUser({
+        electionHash: electionSampleDefinition.electionHash,
+        jurisdiction: DEV_JURISDICTION,
+      }),
+      hasPin: false,
+    },
+  });
+});
+
+test('election setting', async () => {
+  const { apiClient } = setup();
+  // Default election
+  await expect(apiClient.getElection()).resolves.toEqual({
+    title: electionSample.title,
+    path: 'libs/fixtures/data/electionSample.json',
+  });
+
+  await apiClient.setElection({
+    path: 'libs/fixtures/data/electionFamousNames2021/election.json',
+  });
+  await expect(apiClient.getElection()).resolves.toEqual({
+    title: electionFamousNames2021Fixtures.election.title,
+    path: 'libs/fixtures/data/electionFamousNames2021/election.json',
+  });
+
+  // Changing the election should change the election for mocked cards
+  await apiClient.removeCard();
+  await apiClient.insertCard({ role: 'election_manager' });
+  await expect(apiClient.getCardStatus()).resolves.toEqual({
+    status: 'ready',
+    cardDetails: {
+      user: fakeElectionManagerUser({
+        electionHash:
+          electionFamousNames2021Fixtures.electionDefinition.electionHash,
+        jurisdiction: DEV_JURISDICTION,
+      }),
+    },
+  });
+});
+
+// Note: This test overwrites the global mock USB state.
+// The USB mocks don't seem to work on CI, so we skip this test there.
```

Comment:
yeah, I think we might need to add the NO_PASSWORD `sudo` access to CI and even then I don't know if the environment would permit it. need `libs/usb`! good to have a test we can run locally though.